packages/frontend/src/view/components/ExecutionTimeline.tsx

```diff
@@ -0,0 +1,142 @@
+import cx from 'classnames'
+
+type Outcome = 'PENDING' | 'EXECUTED' | 'DISCARDED'
+
+interface Approval {
+  signer: string
+  date: Date
+  method?: string
+}
+
+interface Props {
+  outcome: Outcome
+  approvals: Approval[]
+  submissionDate: Date
+}
+
+export function ExecutionTimeline({
+  outcome,
+  approvals,
+  submissionDate,
+}: Props) {
+  const outcomeVariant =
+    outcome === 'PENDING' ? 'orange' : outcome === 'EXECUTED' ? 'green' : 'red'
+  const outcomeNode = (
+    <TimelineNode variant={outcomeVariant} size="large">
+      <span className={cx('font-semibold', variantToTextColor(outcomeVariant))}>
+        {outcome}
+      </span>
+      {outcomeToDescription(outcome)}
+    </TimelineNode>
+  )
+  const outcomeTimeline = <Timeline variant={outcomeVariant} />
+
+  const approvalNodes = approvals.map((approval, i) => (
+    <>
+      <TimelineNode key={i} variant="orange" size="small">
+        <span className={cx('font-medium', variantToTextColor('orange'))}>
+          Approved
+        </span>
+        {approval.signer}
+        <span className="text-gray-50">
+          {approval.date.toLocaleString()}{' '}
+          {approval.method && `via ${approval.method}`}
+        </span>
+      </TimelineNode>
+      <Timeline variant="orange" />
+    </>
+  ))
+
+  const submissionNode = (
+    <TimelineNode variant="gray" size="large">
+      <div className={cx('font-medium', variantToTextColor('gray'))}>
+        Submitted
+      </div>
+      <div>{submissionDate.toLocaleString()}</div>
+    </TimelineNode>
+  )
+
+  return (
+    <div className="mb-5 flex max-w-fit flex-col items-center">
+      {outcomeNode}
+      {outcomeTimeline}
+      {approvalNodes}
+      {submissionNode}
+    </div>
+  )
+}
+
+type Variant = 'green' | 'orange' | 'gray' | 'red'
+type Size = 'small' | 'large'
+
+function Timeline({ variant }: { variant: Variant }) {
+  return (
+    <div
+      className={cx('h-[52px] w-[1px]', variantToBackgroundColor(variant))}
```

Comment:
```suggestion
      className={cx('h-[52px] w-px', variantToBackgroundColor(variant))}
```

Reply:
[style: size offset in ExecutionTimeline component](https://github.com/l2beat/lz-monitoring/pull/65/commits/3894feb28a519de3a646d8290826f00304c4a0e3)